api/src/opentrons/protocol_engine/state/frustum_helpers.py

```
@@ -286,3 +292,191 @@ def get_well_volumetric_capacity(
+            total_frustum_height=frustum_height,
+        )
+    elif top_cross_section["shape"] == bottom_cross_section["shape"] == "rectangular":
+        frustum_height = height_from_volume_rectangular(
+            volume=target_volume_relative,
+            total_frustum_height=frustum_height,
+            bottom_width=bottom_cross_section["xDimension"],
+            bottom_length=bottom_cross_section["yDimension"],
+            top_width=top_cross_section["xDimension"],
+            top_length=top_cross_section["yDimension"],
+        )
+    return frustum_height
+
+
+def volume_at_height_within_section(
+    top_cross_section: Union[CircularBoundedSection, RectangularBoundedSection],
+    bottom_cross_section: Union[CircularBoundedSection, RectangularBoundedSection],
+    target_height_relative: float,
+    frustum_height: float,
+) -> float:
+    """Calculate a volume within a bounded section according to geometry."""
+    if top_cross_section["shape"] == bottom_cross_section["shape"] == "circular":
+        frustum_volume = volume_from_height_circular(
+            target_height=target_height_relative,
+            total_frustum_height=frustum_height,
+            bottom_radius=(bottom_cross_section["diameter"] / 2),
+            top_radius=(top_cross_section["diameter"] / 2),
+        )
+    elif top_cross_section["shape"] == bottom_cross_section["shape"] == "rectangular":
+        frustum_volume = volume_from_height_rectangular(
+            target_height=target_height_relative,
+            total_frustum_height=frustum_height,
+            bottom_width=bottom_cross_section["xDimension"],
+            bottom_length=bottom_cross_section["yDimension"],
+            top_width=top_cross_section["xDimension"],
+            top_length=top_cross_section["yDimension"],
+        )
+    # else:
+    # TODO(cm): this would be the NEST-96 2uL wells referenced in EXEC-712
+    # we need to input the math attached to that issue
+    return frustum_volume
```

Raise a not implemented instead of a dummy value